docs/source/howto/workflows.rst

```diff
@@ -4,14 +4,217 @@
+Next, the ``define()`` method allows the developer to define the characteristics of the work chain, which are contained in the work chain ``spec``:
+
+* the **inputs**, specified using the ``spec.input()`` method.
+  The first argument of the ``input()`` method is a string that specifies the label of the input, e.g. ``'x'``.
+  The ``valid_type`` keyword argument allows you to specify the required node type of the input.
+  Other keyword arguments allow the developer to set a default for the input, or indicate that an input should not be stored in the database, see :ref:`the process topics section <topics:processes:usage:spec>` for more details.
+* the **outline** or logic of the workflow, specified using the ``spec.outline()`` method.
+  The outline of the workflow is constructed from the methods of the ``WorkChain`` class.
+  For the ``MultiplyAddWorkChain``, the outline is a simple linear sequence of steps, but it's possible to define more complex workflows as well.
+  See the :ref:`work chain outline section <topics:workflows:usage:workchains:define_outline>` for more details.
+* the **outputs**, specified using the ``spec.output()`` method.
+  This method is very similar in its usage to the ``input()`` method.
+* the **exit codes** of the work chain, specified using the ``spec.exit_code()`` method.
+  Exit codes are used to clearly communicate known failure modes of the work chain to the user.
+  The first and second arguments define the ``exit_status`` of the work chain in case of failure (``400``) and the string that the developer can use to reference to the exit code (``ERROR_NEGATIVE_NUMBER``).
+  A descriptive exit message can be provided using the ``message`` keyword argument.
+  For the ``MultiplyAddWorkChain``, we demand that the final result is not a negative number, which is checked in the ``validate_result`` step of the outline.
+
+.. note::
+
+    For more information on the ``define()`` method and the process spec, see the :ref:`corresponding section in the topics <topics:processes:usage:defining>`.
+
+.. code-block:: python
+
+    def multiply(self):
+        """Multiply two integers."""
+        self.ctx.multiple = multiply(self.inputs.x, self.inputs.y)
+
+The ``multiply`` method is the first step in the outline of the ``MultiplyAddWorkChain`` work chain.
+This step simply involves running the calculation function ``multiply``, on the ``x`` and ``y`` **inputs** of the work chain.
+To store the result of this function and use it in the next step of the outline, it is added to the *context* of the work chain using ``self.ctx``.
+
+.. code-block:: python
+
+    def add(self):
+        """Add two numbers with the ArithmeticAddCalculation process."""
+
+        inputs = {'x': self.ctx.multiple, 'y': self.inputs.z, 'code': self.inputs.code}
+        calcjob_node = self.submit(ArithmeticAddCalculation, **inputs)
+
+        return ToContext({'addition': calcjob_node})
```

Comment:
```suggestion
        return ToContext(addition=calcjob_node})
```
Make sure that all snippets match exactly that of the literal-included source code